### PR TITLE
Cloudsiem Quickstart Version 2.0 Updates

### PR DESCRIPTION
**Pull request by himanshu219**
1. Upgrade Python and NodeJS runtime
2. Fixes for CFN lint, CFN Nag and bridgecrew
3. Added versioning 
4. Upgraded SAM applications
5. Upgraded dependencies in lambda helper function for CVE fixes
6. Bug fixes for Cloudtrail collection for bucket path expression
7. Bug fixes for Network Firewall Collection to enable AWS Config and AWS RAM
8. Updated Centralized collection for Security Hub using single configuration policy.

### CODE REVIEW

File: .taskcat.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 project:
-  name: cfn-ps-sumo-logic-cloudsiem
+  name: cfn-ps-sumo-logic-cloudsiem/2
```
Comment (gcasilva) [inline .taskcat.yml:2]: please remove "/2" from project name in taskcat

File: templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml [hunk 2]
```diff
@@ -304,31 +331,39 @@ Resources:
...
-            - !Ref CloudTrailRegions
+            - !Ref CloudTrailRegion
           DeploymentTargets:
             Accounts:
               - !Ref "AWS::AccountId"
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-org.yaml'
-        - S3Region: !Ref "AWS::Region"
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-org.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml:339]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml [hunk 3]
```diff
@@ -304,31 +331,39 @@ Resources:
...
     Type: AWS::CloudFormation::Stack
     Properties:
       Tags:
         - Key: "version"
           Value: !Ref QSVersion
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-app-sumo.yaml'
-        - S3Region: !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-app-sumo.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/cloudtrail-org/cloudtrail-main.yaml:364]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml [hunk 4]
```diff
@@ -471,20 +490,35 @@ Resources:
             Accounts:
               - !Ref SecurityAccountId
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-disassociate-iam-role.yaml'
-        - S3Region:  !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-disassociate-iam-role.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:493]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml [hunk 5]
```diff
@@ -538,9 +572,9 @@ Resources:
             Accounts:
               - !Ref SecurityAccountId
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-sg-policy.yaml'
-        - S3Region:  !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-sg-policy.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:575]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml [hunk 6]
```diff
@@ -574,20 +615,20 @@ Resources:
             Accounts:
               - !Ref SecurityAccountId
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-waf-network-firewall-policy.yaml'
-        - S3Region:  !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-waf-network-firewall-policy.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:618]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml [hunk 7]
```diff
@@ -611,16 +652,18 @@ Resources:
             Accounts:
               - !Ref SecurityAccountId
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-sumo-logsource.yaml'
-        - S3Region:  !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-sumo-logsource.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:655]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml [hunk 8]
```diff
@@ -733,9 +776,9 @@ Resources:
             Accounts:
               - !Ref SecurityAccountId
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-sumo-subscription.yaml'
-        - S3Region:  !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-sumo-subscription.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/firewall-manager-org/firewall-manager-org-master.yaml:779]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml [hunk 9]
```diff
@@ -294,14 +311,14 @@ Resources:
...
-            -  !Select [0, !Split [",", !Ref EnabledRegions]]
+            - !Ref 'AWS::Region'
           DeploymentTargets:
               OrganizationalUnitIds:
                 - !Ref OrganizationRootID
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-delete-detector-role.yaml'
-        - S3Region: !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-delete-detector-role.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml:319]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml [hunk 10]
```diff
@@ -311,6 +328,37 @@ Resources:
...
           ParameterValue: !Ref "AWS::AccountId"
 
+  GuardDutyDeleteDetectorRoleStack:
+    Condition: install_http_logs_source
+    Type: AWS::CloudFormation::Stack
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    Properties:
+      TemplateURL: !Sub
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-delete-detector-role.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml:338]: remove ${QSVersion}

File: templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml [hunk 11]
```diff
@@ -327,14 +375,14 @@ Resources:
...
-        - Regions: !Split [",", !Ref EnabledRegions]
+        - Regions: !Split [",", !GetAtt AvailableServiceRegionsCustomResource.REGIONS]
           DeploymentTargets:
             Accounts:
               - !Ref DelegatedAdminAccountId
       TemplateURL: !Sub
-        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-cloudwatch-event.yaml'
-        - S3Region:  !Ref 'AWS::Region'
-          S3Bucket: !Sub '${QSS3BucketName}-${AWS::Region}'
+        - 'https://${S3Bucket}.s3.${S3Region}.${AWS::URLSuffix}/${QSS3KeyPrefix}${QSVersion}/templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-cloudwatch-event.yaml'
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/guardduty-org/guardduty-org-configuration.yaml:383]: remove ${QSVersion}

**Review by gcasilva — COMMENTED** [review]
Please remove ${QSVersion} from Template URLs, check in the comments.
Also remove /2 from project name in taskcat.yml, check in comments.

File: .taskcat.yml (hunk 1 above)
Comment (gcasilva) [reply .taskcat.yml:2]: Also uncomment and change lines 19-22 so we can use in the automated tests by referencing SSM Parameter Store for these variables in our test AWS Account:

Section1SumoLogicDeployment: "us2"
Section1SumoLogicAccessID: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessID]
Section1SumoLogicAccessKey: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessKey]
Section1SumoLogicOrganizationId: $[taskcat_ssm_/sumologic/Section1SumoLogicOrganizationId]

File: .taskcat.yml [hunk 12]
```diff
@@ -12,44 +12,61 @@ tests:
     regions:
     - us-east-2
     parameters:
-      QSS3KeyPrefix: "quickstart-sumo-logic-cloudsiem/"
+      QSS3KeyPrefix: "cfn-ps-sumo-logic-cloudsiem/"
       QSS3BucketName:  "$[taskcat_autobucket_prefix]"
       QSS3BucketRegion: "us-east-2"
```
Comment (gcasilva) [inline .taskcat.yml:17]: change to us-east-1 for our automated tests.

File: .taskcat.yml [hunk 13]
```diff
@@ -12,44 +12,61 @@ tests:
...
-      # Section1SumoLogicAccessID: ""
-      # Section1SumoLogicAccessKey: ""
-      # Section1SumoLogicOrganizationId: ""
+      QSVersion: "2"
+      Section1SumoLogicDeployment: "us2"
+      Section1SumoLogicAccessID: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessID]
+      Section1SumoLogicAccessKey: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessKey]
+      Section1SumoLogicOrganizationId: $[taskcat_ssm_/sumologic/Section1SumoLogicOrganizationId]
       Section1SecurityToolingAccountId: ""
       Section1ToolingAndLoggingRegion: "us-east-2"
```
Comment (gcasilva) [inline .taskcat.yml:24]: change to us-east-1 for our automated tests.

File: .taskcat.yml [hunk 14]
```diff
@@ -12,44 +12,61 @@ tests:
...
-      Section2InstallSumoCloudSecurityMonitoringandAnalyticsGuardDutyApp: "No"
+      Section2InstallSumoCloudSecurityMonitoringandAnalyticsGuardDutyApp: "Yes"
       Section2GuardDutyRegions: "us-east-2"
+      Section2AdditionalConfigurationFeatures: "MalwareProtection,S3Logs"
       Section2GuardDutyCreateHttpLogsSource: "Yes"
-      Section3CloudTrailRegions: "us-east-2"
-      Section3InstallAWSCloudTrail: "Yes"
+      Section2GuardDutyHttpLogsSourceCategoryName: "aws/quickstart/guardduty/logs"
+
+      Section3CloudTrailRegion: "us-east-2"
```
Comment (gcasilva) [inline .taskcat.yml:37]: change to us-east-1 for our automated tests.

File: .taskcat.yml [hunk 15]
```diff
@@ -12,44 +12,61 @@ tests:
...
       Section3InstallCloudTrailSecOpsApp: "Yes"
       Section3CloudTrailCreateS3LogsSource: "Yes"
       Section3CreateCloudTrailS3Bucket: "Yes"
-
+      Section3DisassociateAdminAccountOnDeleteStack: "false"
+      Section3CloudTrailLogsSourceCategoryName: "aws/quickstart/cloudtrail/logs"
+      Section3CloudTrailBucketPathExpression: "CloudTrail/AWSLogs/*"
+      
       Section4SecurityHubEnableSecurityHub: "Yes"
+      Section4SecurityHubAggregationRegion: "us-east-2"
```
Comment (gcasilva) [inline .taskcat.yml:51]: change to us-east-1 for our automated tests.

File: .taskcat.yml [hunk 16]
```diff
@@ -12,44 +12,61 @@ tests:
...
-      Section4SecurityHubRegionsToEnable: "us-east-2"
       Section4SecurityHubCreateHttpLogsSource: "Yes"
+      Section4SecurityHubHttpLogsSourceCategoryName: "aws/quickstart/securityhub/logs"
+      Section4SecurityHubSecurityStandards: "FSBP_v1.0.0,CIS_v1.2.0,PCI_DSS_v3.2.1,NIST_v5.0.0,CIS_v1.4.0"
 
       Section6EnableFirewallManager: "Yes"
       Section6DisassociateAdminAccountOnDeleteStack: "false"
       Section6InstallSumoAWSWAFCloudSecurityMonitoringAndAnalyticsApp: "Yes"
       Section6InstallSumoAWSNetworkFirewallApp: "Yes"
       Section6FirewallManagerPolicyRegions: "us-east-2"
```
Comment (gcasilva) [inline .taskcat.yml:63]: change to us-east-1 for our automated tests.

**Comment by gcasilva** [comment]
@himanshu219 please change all regions in .taskcat.yml to us-east-1 for our automated tests to execute properly in our test account.

**Comment by gcasilva** [comment]
@himanshu219 can you please confirm if there's any additional commits to be added to this PR before we do final review/tests?

**Comment by himanshu219** [comment]
> @himanshu219 can you please confirm if there's any additional commits to be added to this PR before we do final review/tests?

There might be some commits related to documentation.

**Comment by gcasilva** [comment]
@himanshu219 please update me here when all commits have been sent so I can do final review and test. Thanks

**Comment by himanshu219** [comment]
@gcasilva  yes all the commits are sent for both the quickstart PRs

File: templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml [hunk 17]
```diff
@@ -511,185 +492,80 @@ Resources:
...
-          ENABLE_PCI_STANDARD: !Sub ${EnablePCIStandard}
-          PCI_STANDARD_VERSION: !Sub ${PCIStandardVersion}
-          ENABLE_SBP_STANDARD: !Sub ${EnableSBPStandard}
-          SBP_STANDARD_VERSION: !Sub ${SBPStandardVersion}
-          REGIONS_TO_ENABLE: !Sub ${RegionsToEnable}
-          CONTROL_TOWER_REGIONS_ONLY: !Sub ${ControlTowerRegionsOnly}
-          ENABLE_PROWLER_INTEGRATION: !Sub ${EnableProwlerIntegration}
-
-  SecurityHubEnablerLambdaCustomResource:
+  SecurityHubLambdaCustomResource:
```
Comment (gcasilva) [inline templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @himanshu219 
I'm getting an error on this resource, it seems like the policy for the IAM Role for this resource needs to be updated to add UpdateOrganizationConfiguration API from Security Hub, please check and fix:
"Received response status [FAILED] from custom resource. Message returned: An error occurred (AccessDeniedException) when calling the UpdateOrganizationConfiguration operation: You don't have permissions to access this resource."

**Review by gcasilva — COMMENTED** [review]
@himanshu219 I'm having an error on SecurityHubStack stack please check my comment to review/fix.

File: templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml (hunk 17 above)
Comment (langmtt) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @gcasilva Please share the values of parameters: Section4SecurityHubAggregationRegion, Section4SecurityHubLinkedRegions

Comment (gcasilva) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @langmtt Below are the value of the parameters. FYI these regions are also correctly enabled as governed regions in Control Tower.
Section4SecurityHubAggregationRegion: us-east-1
Section4SecurityHubLinkedRegions: us-east-2, us-west-1

Comment (gcasilva) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @himanshu219 Tested successfully the deployment with the fix sent. Can you please confirm if there's any additional documentation changes you'll add to this PR before we close it and merge it to main?
Thanks

Comment (himanshu219) [reply templates/apps/sumo-aws-apps/security-hub-org/security-hub-org-main.yaml:549]: @gcasilva  nops, there is no doc changes required.

**Review by gcasilva — DISMISSED** [review]

File: .taskcat.yml [hunk 18]
```diff
@@ -3,53 +3,70 @@ project:
...
-      # Section1SumoLogicAccessID: ""
-      # Section1SumoLogicAccessKey: ""
-      # Section1SumoLogicOrganizationId: ""
+      QSS3BucketRegion: "us-east-1"
+      QSVersion: "2"
+      Section1SumoLogicDeployment: "us2"
+      Section1SumoLogicAccessID: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessID]
+      Section1SumoLogicAccessKey: $[taskcat_ssm_/sumologic/Section1SumoLogicAccessKey]
+      Section1SumoLogicOrganizationId: $[taskcat_ssm_/sumologic/Section1SumoLogicOrganizationId]
       Section1SecurityToolingAccountId: ""
```
Comment (gcasilva) [inline .taskcat.yml:23]: @langmtt please change this to the below so we can execute the automated tests to merge:
Section1SecurityToolingAccountId: $[taskcat_ssm_/sumologic/Section1SecurityToolingAccountId]

File: .taskcat.yml [hunk 19]
```diff
@@ -3,53 +3,70 @@ project:
...
-      Section1ToolingAndLoggingRegion: "us-east-2"
+      Section1ToolingAndLoggingRegion: "us-east-1"
       Section1LogArchivingAccountId: ""
```
Comment (gcasilva) [inline .taskcat.yml:25]: @langmtt please change this to the below so we can execute the automated tests to merge:
Section1LogArchivingAccountId: $[taskcat_ssm_/sumologic/Section1LogArchivingAccountId]

File: .taskcat.yml [hunk 20]
```diff
@@ -3,53 +3,70 @@ project:
...
       Section1OrganizationRootID: ""
```
Comment (gcasilva) [inline .taskcat.yml:26]: @langmtt please change this to the below so we can execute the automated tests to merge:
Section1OrganizationRootID: $[taskcat_ssm_/sumologic/Section1OrganizationRootID]

**Comment by himanshu219** [comment]
@gcasilva please let us know if there are any action items on us for these failing e2e tests.

**Comment by gcasilva** [comment]
> @gcasilva please let us know if there are any action items on us for these failing e2e tests.
@himanshu219: please fix the below in taskcat.yml 
Section6DeliveryStreamName is failing in the automated test due to length validation in the template (16 character max). Please change the value to 'sumologic' so it matches the default in the template

**Review by gcasilva — APPROVED** [review]

**Review by langmtt — APPROVED** [review]

File: .taskcat.yml [hunk 21]
```diff
@@ -3,53 +3,70 @@ project:
...
 
       Section6EnableFirewallManager: "Yes"
       Section6DisassociateAdminAccountOnDeleteStack: "false"
       Section6InstallSumoAWSWAFCloudSecurityMonitoringAndAnalyticsApp: "Yes"
       Section6InstallSumoAWSNetworkFirewallApp: "Yes"
-      Section6FirewallManagerPolicyRegions: "us-east-2"
+      Section6FirewallManagerPolicyRegions: "us-east-1"
       Section6CreateDeliveryStreamSource: "Yes"
-      Section6DeliveryStreamName: $[taskcat_random-string]
+      Section6DeliveryStreamName: sumologic-kinesis-$[taskcat_random-string]
```
[the remark this answers is not in the document]
Comment (himanshu219) [reply .taskcat.yml]: @gcasilva  is it still failing due to length `sumo-kinesis` is 12 chars long ?

[the remark this answers is not in the document]
Comment (himanshu219) [reply .taskcat.yml]: @gcasilva  any update on above ? we have verified by testing with taskcat also and locally as well. It seems to be passing for us. Could you please let us know why the e2e-tests are failing, is it with same error or some different error?

**Review by gcasilva — CHANGES_REQUESTED** [review]
@himanshu219 @langmatt please check review for changes needed due to e2e failing

**Comment by gcasilva** [comment]
@himanshu219 I'll merge this manually tomorrow with the AWS Quickstarts team and I can work on fixing whatever is needed for the e2e deployment later since the template is working appropriately from customer perspective. Thanks

**Comment by langmatt** [comment]
Am I supposed to be on this?

**Review by gcasilva — APPROVED** [review]

**Pull request merged**